File: apps/mybookkeeper/backend/app/schemas/welcome_manuals/welcome_manual_create_request.py
"""Pydantic schema for POST /welcome-manuals request body.

``organization_id`` and ``user_id`` are NOT accepted — they're resolved
server-side from the request context. ``property_id`` is an optional tag.
"""
from __future__ import annotations

import uuid

from pydantic import BaseModel, ConfigDict, Field

from app.core.welcome_manual_constants import WELCOME_MANUAL_TITLE_MAX_LEN


class WelcomeManualCreateRequest(BaseModel):
    title: str = Field(min_length=1, max_length=WELCOME_MANUAL_TITLE_MAX_LEN)
    intro_text: str | None = None
    property_id: uuid.UUID | None = None
    # When True (default), the new manual is pre-seeded with stub sections
    # (Wi-Fi, Parking, Trash & Recycling, Laundry, Check-out) so the host
    # starts from a structure instead of a blank page.
    seed_default_sections: bool = True

    model_config = ConfigDict(extra="forbid")

File: apps/mybookkeeper/frontend/src/shared/types/welcome-manual/welcome-manual-create-request.ts
/**
 * Body for POST /welcome-manuals. ``organization_id``/``user_id`` are resolved
 * server-side. ``seed_default_sections`` defaults to true on the backend, but
 * the create dialog always sends it explicitly.
 */
export interface WelcomeManualCreateRequest {
  title: string;
  intro_text?: string | null;
  property_id?: string | null;
  seed_default_sections: boolean;
}
